lib/active_merchant/billing/gateways/stripe_payment_intents.rb

```diff
@@ -299,7 +320,34 @@ def extract_token_from_string_and_maybe_add_customer_id(post, payment_method)
           post[:customer] = customer_id
         end
 
-        post[:payment_method] = payment_method
+        if payment_method.include?('tok_')
+          add_payment_method_card_data_tok(post, payment_method)
+        else
+          post[:payment_method] = payment_method
+        end
+      end
+
+      def tokenize_apple_google(payment, options = {})
+        tokenization_method = options[:tokenization_method] || 'android_pay'
+        tokenization_method = 'apple_pay' if payment.source.to_sym == :apple_pay
```

Comment:
🛑   Warning:

We need a better logic here, seems to be the case that tokenization_method param can be determined on the options or on the payment object but is not clear the priority order, so I have some questions:

1) what is the logic behind that couple of possibilities ? my understanding is the `payment.source` should be only truth. 
2) What happens if options[:tokenization_method] maybe has the value "google_pay" ?